wremnants/include/theory_corrections.h

```diff
@@ -17,11 +18,11 @@ typedef typename T::storage_type::value_type::tensor_t tensor_t;
     TensorCorrectionsHelper3D(T&& corrections) :
         correctionHist_(std::make_shared<const T>(std::move(corrections))) {}
 
-    tensor_t operator() (double x, double y, double z) {
+    tensor_t operator() (double x, double y, double z, double nominal_weight = 1.0) {
         auto xbin = correctionHist_->template axis<0>().index(x);
         auto ybin = correctionHist_->template axis<1>().index(y);
         auto zbin = correctionHist_->template axis<2>().index(z);
-        return correctionHist_->at(xbin, ybin, zbin).data();
+        return nominal_weight*correctionHist_->at(xbin, ybin, zbin).data();
```

Comment:
We still need to add an option to to actually include correction_weight[0] in the nominal weight and then divide it out for the systematic variations

Reply:
Ok yes good point.  I think like this is still an improvement though, since then at least the alternate histogram does contain the consistent product of weights.